src/python/schema.rs

```diff
@@ -0,0 +1,66 @@
+use pyo3::prelude::*;
+
+use crate::datatypes;
+use crate::python::datatype;
+use crate::schema;
+
+#[pyclass]
+pub struct PySchema {
+    pub schema: schema::SchemaRef,
+}
+
+#[pyclass]
+pub struct PyField {
+    pub field: datatypes::Field,
+}
+
+#[pymethods]
+impl PySchema {
+    pub fn __getitem__(&self, name: &str) -> PyResult<PyField> {
+        Ok(self.schema.get_field(name)?.clone().into())
+    }
+
+    pub fn names(&self) -> PyResult<Vec<String>> {
+        Ok(self.schema.names()?)
+    }
+
+    pub fn union(&self, other: &PySchema) -> PyResult<PySchema> {
```

Comment:
this union method should be implemented in `Schema` rather than having the logic in the pybindings